lib/vhost/vhost_user.c

```diff
@@ -342,6 +344,35 @@ vhost_user_get_features(struct virtio_net **pdev,
 	return RTE_VHOST_MSG_RESULT_REPLY;
 }
 
+static int
+vhost_user_get_config(struct virtio_net **pdev,
+			struct vhu_msg_context *ctx,
+			int main_fd __rte_unused)
+{
+	struct virtio_net *dev = *pdev;
+	struct rte_vdpa_device *vdpa_dev;
+
+	if (validate_msg_fds(dev, ctx, 0) != 0)
+		return RTE_VHOST_MSG_RESULT_ERR;
+
+	vdpa_dev = dev->vdpa_dev;
+
+	if (vdpa_dev)
+		vdpa_dev->ops->get_dev_config(dev->vid,ctx->msg.payload.cfg.region,ctx->msg.payload.cfg.size);
+
+	return RTE_VHOST_MSG_RESULT_REPLY;
+}
+
+static int
+vhost_user_set_config(struct virtio_net **pdev __rte_unused,
+			struct vhu_msg_context *ctx __rte_unused,
+			int main_fd __rte_unused)
+{
+    VHOST_LOG_CONFIG(ERR, "vhost_user_set_config not supported");
```

Comment:
in the commit msg you say, it is must, but in the actual code it does nothing.

How does it work?
When set_config() called?
Does the caller ignore the return status?

if so, can this routine be set to NULL?

What all config fields are setup today by QEMU on this area?

Reply:
set_config is read only for virtio-blk device, so, it should not be called.